superset/models/core.py

```diff
@@ -415,7 +415,11 @@ def get_df(  # pylint: disable=too-many-locals
         mutator: Optional[Callable[[pd.DataFrame], None]] = None,
     ) -> pd.DataFrame:
         sqls = self.db_engine_spec.parse_sql(sql)
+
         engine = self.get_sqla_engine(schema)
+        username = utils.get_username() or username
```

Comment:
This variable doesn't appear to be defined in this scope?
```suggestion
        username = utils.get_username()
```